### ISSUE TITLE
Dockerizing the app

### ISSUE THREAD

**Issue by TestCandidate** [bntjah/lancache#60]
Let's make this project a docker image so the setup would be pull and play :) 
I think I can work on it if there's any interest?

**Comment by bntjah** [comment]
My docker experience is limited to 1 to none images :P
As I haven't gotten around to playing with it I would say go nuts :-)

2017-10-19 12:14 GMT+02:00 TestCandidate <notifications@github.com>:

**Comment by TestCandidate** [comment]
Well, Mine is a 2 :) but I'm gonna mess with it and see where this leads
Will get back soon with an update on this thread.

Just a question, would you consider the installer script fully functional? or do you recommend against using it? because it would be helpful with the dockerizing process.

thanks

**Comment by bntjah** [comment]
It was a starter for me to get working within bash so it needs improving
but you could use it as a base I imagine

2017-10-19 12:31 GMT+02:00 TestCandidate <notifications@github.com>:

**Comment by travispk** [comment]
The software is pretty simplistic in regards to just running the installer and bam it's done.  But I understand dockerizing it might be worthwhile.   As well, if we can complete #29 that would greatly simplify both install ways, as we don't have to mess with the OS much at all, especially networking.

**Comment by muffeeee** [comment]
no thanks, one of the reasons why I chose this was because it wasn't already dockerized.

**Comment by TestCandidate** [comment]
@muffeeee  could you elaborate? why not?

**Comment by travispk** [comment]
I suggest both, the autoinstall script is awesome.  I think the install could be even more simplified as I listed before.  But as well, there could be a docker version as well.  Seems like a lot of apps are going that direction.

But yeah, I would curious to hear explicitly why not to do it.

**Comment by muffeeee** [comment]
@TestCandidate because doing anything in regards of modifying something that's dockerized is generally a pain? Anyway, I'm not the biggest fan of docker.

I've had to done multiple adjustments to lancache such as adding/removing cached services and doing custom setups to match the connectivity that was provided. I've also spread the vhosts on multiple servers as an easy way to balance the load, as steam alone is more active than the rest of the services and resources are sometimes limited. +1 @travispk, having options would be greatly appreciated.

**Comment by TestCandidate** [comment]
Exactly. this won't be the only method. but for people who aren't technical and don't want any headaches, it would be as easy as ship the container, plug and play. anyway, I still didn't have the time to start working on this. next month hopefully.

**Comment by cyrianox** [comment]
Somebody already test docker performance against local server ?

Le sam. 28 oct. 2017 à 19:14, TestCandidate <notifications@github.com> a
écrit :

**Comment by TestCandidate** [comment]
I didn't. But it shouldn't make any difference. In theory

**Comment by soulflyman** [comment]
Hi I run an older installation of the lancache setup  at home for only two clients and a installation in youth center with 5-7 clients. The Internet connection has a really limited bandwidth and lancache was a blessing.

Fact is that I'm not a sys admin guy and setting it up costed me a lot of time back then. Updating is for me nearly impossible with my limited time and also hard to to because it runs on bind and bind is no longer in use.

Long story short. Running a docker image on the local QNAP NAS would be a pleasing for us and it could be easier to update.
But this does not mean that docker is a must have for me. I would rather favor a solution with a really simple update possibility, a script easy as running 'apt update & apt upgrade'. 

By the way, thanks for all your work. You are awesome and lancache helped us a lot!

**Comment by BloodyIron** [comment]
I'm not against docker stuff, but I for one appreciate that I can set this up manually. There is another docker image project that I think may work for you.

But I myself am explicitly interested in being able to set this up manually, as I want to maintain control, and keep performance very high. Docker, for me, would get in the way of that.

**Comment by muffeeee** [comment]
I totally agree with @BloodyIron.

If someone wants to set up a fork of lancache with an updated docker image, go ahead. It'll helpa lot of inexperienced users. However, my own experience with lancache is that it requires a lot of maintenance and troubleshooting as soon as you reach more than ~50 users and on >1Gbitps links (using LACP/Link aggregation).

These issues have mainly been linked to DNS and networking on our cache node(s). I'm working on compiling a list of all the issues we've had (and how to solve them) next week, and hopefully making a pull request on it as well.

IMO, the best approach would be to make a webpanel for lancache as well as reducing the number of IP adresses. Having a simple setup wizard and a webpanel frontend to manage lancache would make this a more feasible approach to even small LAN-parties who don't have the time to manage servers..

(Sorry if I'm writing gibberish. I'm tired as f***.)

**Comment by elico** [comment]
@muffeeee What is your main use for lancache?
Do you have a specific targeted downloads you wish to cache?

**Comment by elico** [comment]
@bntjah @muffeeee I have created a basic MS updates lancache Docker containers at: 
- <https://github.com/elico/lancache-web-alpine>
- <https://github.com/elico/lancache-web-ubuntu>

Since it's a container it has a single IP address..
My assumption is that it doesn't need more then one IP to work in most setups.
If you are an ISP that has 50k clients... this solution should be fine tuned for your environment.
